NWN.Anvil/src/main/API/Extensions/StringExtensions.cs

```diff
@@ -108,6 +109,42 @@ public static bool ParseIntBool(this string intBoolString, bool defaultValue)
       return uint.Parse(objectIdString, NumberStyles.HexNumber).ToNwObject<T>();
     }
 
+    /// <summary>
+    /// Tries to resolve the specified GameObject ID string to an object. A return value
+    /// indicates whether the conversion succeeded or failed.<br/>
+    /// This is the temporary ID created from <see cref="NwObject.ToString"/>. See <see cref="GuidExtensions.ToNwObject"/> to parse persistent UUIDs.
+    /// </summary>
+    /// <param name="objectIdString">The object ID string to parse.</param>
+    /// <param name="result">When this method returns, contains the object referenced by
+    /// the number contained in objectIdString, if the conversion succeeded, or null if
+    /// the conversion failed. The conversion fails if the s parameter is null or System.String.Empty,
+    /// is not in a format compliant with style, or represents an invalid object reference.</param>
+    /// <returns>true if objectIdString was converted successfully; otherwise, false.</returns>
+    public static bool TryParseObject(this string objectIdString, [NotNullWhen(true)] out NwObject? result)
+    {
+      if (uint.TryParse(objectIdString, NumberStyles.HexNumber, null, out uint res) && res.ToNwObject() is NwObject obj)
```

Comment:
```suggestion
      if (uint.TryParse(objectIdString, NumberStyles.HexNumber, null, out uint res) && res.ToNwObject() is {} obj)
```
Super small nit pick, the syntax it is looking for is this.

Reply:
Oh, that's new to me. What's the argument for using `is {}` over `is NwObject`? I thought both checked for `is not null` but that `is NwObject` also did type narrowing.

Reply:
Since `ToNwObject()` already returns `NwObject` as a type, it sees `is NwObject` as a redundant check, as the compiler already knows that it is an NwObject.

You only do this so you can assign the object to a variable, without needing to do it in a nested call. I added a similar extension method as a helper in Toolbox, as it is really handy for setting up Nui views: https://github.com/Jorteck/NWN.Toolbox/blob/main/NWN.Toolbox/src/Extensions/NuiUtils.cs#L31-L35
https://github.com/Jorteck/NWN.Toolbox/blob/main/NWN.Toolbox/src/Features/ToolWindows/ObjectTools/ChatImpersonatorWindowView.cs#L55

I'd argue that it is valid as you are checking a null type vs a not null type, but I guess the analyser sees it as the same type. Perhaps there is a tiny optimisation to the approach suggested by the analyser as you are checking for null only, rather than also the type which we already know is an NwObject (or the specified generic type).

Reply:
Ah! The type is already known! Clever compiler :) Thank you for the explanation, it should be fixed now